http/src/request/validate.rs

```diff
@@ -310,99 +302,79 @@ pub fn embed(embed: &Embed) -> Result<(), EmbedValidationError> {
     Ok(())
 }
 
-pub fn get_audit_log_limit(value: u64) -> bool {
+pub const fn get_audit_log_limit(value: u64) -> bool {
     // <https://discordapp.com/developers/docs/resources/audit-log#get-guild-audit-log-query-string-parameters>
-    (1..=100).contains(&value)
+    value >= 1 && value <= 100
 }
 
-pub fn get_channel_messages_limit(value: u64) -> bool {
+pub const fn get_channel_messages_limit(value: u64) -> bool {
     // <https://discordapp.com/developers/docs/resources/channel#get-channel-messages-query-string-params>
-    (1..=100).contains(&value)
+    value >= 1 && value <= 100
 }
 
-pub fn get_current_user_guilds_limit(value: u64) -> bool {
+pub const fn get_current_user_guilds_limit(value: u64) -> bool {
     // <https://discordapp.com/developers/docs/resources/user#get-current-user-guilds-query-string-params>
-    (1..=100).contains(&value)
+    value >= 1 && value <= 100
 }
 
-pub fn get_guild_members_limit(value: u64) -> bool {
+pub const fn get_guild_members_limit(value: u64) -> bool {
     // <https://discordapp.com/developers/docs/resources/guild#list-guild-members-query-string-params>
-    (1..=1000).contains(&value)
+    value >= 1 && value <= 1000
 }
 
-pub fn search_guild_members_limit(value: u64) -> bool {
+pub const fn search_guild_members_limit(value: u64) -> bool {
     value > 0 && value <= 1000
 }
 
-pub fn get_reactions_limit(value: u64) -> bool {
+pub const fn get_reactions_limit(value: u64) -> bool {
     // <https://discordapp.com/developers/docs/resources/channel#get-reactions-query-string-params>
-    (1..=100).contains(&value)
+    value >= 1 && value <= 100
 }
 
-pub fn guild_name(value: impl AsRef<str>) -> bool {
-    _guild_name(value.as_ref())
-}
-
-fn _guild_name(value: &str) -> bool {
+pub fn guild_name(value: &str) -> bool {
     let len = value.chars().count();
 
     // <https://discordapp.com/developers/docs/resources/guild#guild-object-guild-structure>
     (2..=100).contains(&len)
 }
 
-pub fn guild_prune_days(value: u64) -> bool {
+pub const fn guild_prune_days(value: u64) -> bool {
     // <https://discordapp.com/developers/docs/resources/guild#get-guild-prune-count-query-string-params>
     value > 0 && value <= 30
 }
 
-pub fn invite_max_age(value: u64) -> bool {
+pub const fn invite_max_age(value: u64) -> bool {
     // <https://discord.com/developers/docs/resources/channel#create-channel-invite-json-params>
     value <= 604_800
 }
 
-pub fn invite_max_uses(value: u64) -> bool {
+pub const fn invite_max_uses(value: u64) -> bool {
     // <https://discord.com/developers/docs/resources/channel#create-channel-invite-json-params>
     value <= 100
 }
 
-pub fn nickname(value: impl AsRef<str>) -> bool {
-    _nickname(value.as_ref())
-}
-
-fn _nickname(value: &str) -> bool {
+pub fn nickname(value: &str) -> bool {
```

Comment:
Should the changes like be included here, they seem a bit unrelated

Reply:
Reverted